.ibm/pipelines/openshift-tests.sh

```diff
@@ -153,13 +153,34 @@ run_tests() {
 }
 
 check_backstage_running() {
-  # Check if Backstage is up and running
-  BACKSTAGE_URL_RESPONSE=$(curl -Is "https://${RELEASE_NAME}-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}" | head -n 1)
+  local url="https://${RELEASE_NAME}-backstage-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}"
+  
+  # Maximum number of attempts to check URL
+  local max_attempts=30
+  # Time in seconds to wait
+  local wait_seconds=30
+
+  echo "Checking if Backstage is up and running at $url"
+
+  for ((i=1; i<=max_attempts; i++)); do
+    # Get the status code
+    local http_status=$(curl -o /dev/null -s -w "%{http_code}\n" "$url")
+
+    # Check if the status code is 200
+    if [[ $http_status -eq 200 ]]; then
```

Comment:
you might also get a 302 if any redirection is happening... maybe it's better to test that you're NOT getting a 404? 

Also you could do a `curl -I` to get just the http headers, then more easily skip that for the reponse code. 

```
$➔ curl -sSI https://github.com/janus-idp/helm-backstage/releases/download/backstage-2.10.1/backstage-2.10.1.tgz | grep HTTP
HTTP/2 302 
```

Just an idea!